### DISCUSSION TITLE
ipsec vpn between AWS strongswan EC2 and juniper srx won`t establish

### DISCUSSION THREAD

**Discussion by msstrong** [strongswan/strongswan#238] [Q&A]
Hi all ,
i need your help regarding this strange issue:
i`m running from our side a strongswan machine on an EC2 instance to connect with a juniper srx from customer side 
However the vpn fails to go up as it is sending vpn traffic through port 4500 although no nat-t is being needed or configured.
for some reason the EC2 see itself as behind a nat device.
Here are the logs messages i see:

**Mar  8 14:15:40 ip-172-28-16-214 charon: 16[IKE] giving up after 5 retransmits
Mar  8 14:15:40 ip-172-28-16-214 charon: 16[IKE] peer not responding, trying again (3/3)
Mar  8 14:15:40 ip-172-28-16-214 charon: 16[IKE] initiating IKE_SA sibs-vpn[1] to 195.138.11.110
Mar  8 14:15:40 ip-172-28-16-214 charon: 16[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Mar  8 14:15:40 ip-172-28-16-214 charon: 16[NET] sending packet: from 172.28.16.214[500] to 195.138.11.110[500] (1108 bytes)
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[NET] received packet: from 195.138.11.110[500] to 172.28.16.214[500] (282 bytes)
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N((40002)) V ]
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[ENC] received unknown vendor ID: 69:93:69:22:87:41:c6:d4:ca:09:4c:93:e2:42:c9:de:19:e7:b7:c6:00:00:00:05:00:00:05:00
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/ECP_256
Mar  8 14:10:10 ip-172-28-16-214 charon: 09[IKE] local host is behind NAT, sending keep alives
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[IKE] authentication of '18.198.76.42' (myself) with pre-shared key
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[IKE] establishing CHILD_SA sibs-vpn{3}
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[ENC] generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr AUTH CPRQ(ADDR DNS) SA TSi TSr N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
Mar  8 14:15:40 ip-172-28-16-214 charon: 05[NET] sending packet: from 172.28.16.214[4500] to 195.138.11.110[4500] (368 bytes)
Mar  8 14:15:44 ip-172-28-16-214 charon: 08[IKE] retransmit 1 of request with message ID 1
Mar  8 14:15:44 ip-172-28-16-214 charon: 08[NET] sending packet: from 172.28.16.214[4500] to 195.138.11.110[4500] (368 bytes)
Mar  8 14:15:51 ip-172-28-16-214 charon: 09[IKE] retransmit 2 of request with message ID 1
Mar  8 14:15:51 ip-172-28-16-214 charon: 09[NET] sending packet: from 172.28.16.214[4500] to 195.138.11.110[4500] (368 bytes)
Mar  8 14:16:04 ip-172-28-16-214 charon: 10[IKE] retransmit 3 of request with message ID 1
Mar  8 14:16:04 ip-172-28-16-214 charon: 10[NET] sending packet: from 172.28.16.214[4500] to 195.138.11.110[4500] (368 bytes)
Mar  8 14:16:28 ip-172-28-16-214 charon: 12[IKE] retransmit 4 of request with message ID 1
Mar  8 14:16:28 ip-172-28-16-214 charon: 12[NET] sending packet: from 172.28.16.214[4500] to 195.138.11.110[4500] (368 bytes)
Mar  8 14:17:10 ip-172-28-16-214 charon: 14[IKE] retransmit 5 of request with message ID 1
Mar  8 14:17:10 ip-172-28-16-214 charon: 14[NET] sending packet: from 172.28.16.214[4500] to 195.138.11.110[4500] (368 bytes)
Mar  8 14:18:25 ip-172-28-16-214 charon: 05[KNL] creating delete job for CHILD_SA ESP/0xcaf6b6c7/172.28.16.214
Mar  8 14:18:25 ip-172-28-16-214 charon: 05[JOB] CHILD_SA ESP/0xcaf6b6c7/172.28.16.214 not found for delete
Mar  8 14:18:25 ip-172-28-16-214 charon: 07[IKE] giving up after 5 retransmits
Mar  8 14:18:25 ip-172-28-16-214 charon: 07[IKE] establishing IKE_SA failed, peer not responding**

here is ipsec.conf  file:

conn ipsec-vpn # Name of the connection.
    type=tunnel
    authby=secret
    auto=start # load connection and initiate it on startup
    keyexchange=ikev2
    mobike=no
    #forceencaps=no

# Network Information
    left=172.28.16.214
    leftsourceip=172.28.16.214
    leftid=18.198.76.42 # Elastic IP of the VPN instance
    leftsubnet=172.28.16.208/28 # If you have multiple subnets, use "leftsubnets = {172.31.1.0/24 172.31.3.0/24 [...]}"
    #leftnexthop=%defaultroute
    right=195.138.11.110 # SIX VPN peer public IP address
    rightid=195.138.11.110  # SIX VPN peer ID
    rightsubnet=192.168.0.0/16 # IP range of SIX network. Use "rightsubnets" if multiple subnets
    mobike=no
    # Security Info
    ike=aes256-sha256-ecp256 # IKE Encryption Policy and Diffie-Hallman Group
    ikelifetime=28800s # IKE Lifetime
    esp=aes256-sha2_256-ecp256 # ESP Encryption policy and Diffie-Hallman Group
    salifetime=1800s # IPSec Lifetime

and here is the ipsec.secret file:

18.198.76.42 195.138.11.110 : PSK "password"
i really don`t understand why the local host "the ec2" is behind nat although there is no nat device in the middle 

thank you in advance

**Comment by Thermi** [comment]
NAT is detected so NAT-T is enabled.
Enable NAT-T and UDP encapsulation on the SRX.
EC2 assigns private IPs to the instances. The public IPs are part of a NAT gateway in the VPC.

**Reply by msstrong** [reply]
@Thermi there is no nat-T in my setup and it is not possible to  enable it from customer side as both peers are not behind nat .i did the exact same setup few weeks ago but with ikev1 and i did not have this bahavior thats why i`m confused  .Also there is no nat gateway in my vpc ,i have one public subnet in my vpc so the internet traffic is going through internet gateway and the public IP here is an Elastic IP address.but thanks anyway

**Comment by Thermi** [comment]
Your host uses 172.28.16.214 and that is squarely in 172.16.0.0/12, a private, non-routable subnet.
For sure there is some NAT gateway in between.

Am March 9, 2021 6:55:50 PM UTC schrieb msstrong <notifications@github.com>:
>@Thermi there is no nat-T in my setup and it is not possible to  enable
>it from customer side as both peers are not behind nat .i did the exact
>same setup few weeks ago but with ikev1 and i did not have this
>bahavior thats why i`m confused  .Also there is no nat gateway in my
>vpc ,i have one public subnet in my vpc so the internet traffic is
>going through internet gateway and the public IP here is an Elastic IP
>address.but thanks anyway
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/238#discussioncomment-456647

-- 
Sent from mobile
